### PR TITLE
log2 op

### PR DESCRIPTION
**Pull request by aarushjain29**

### CODE REVIEW

File: src/include/migraphx/op/log2.hpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct log2 : unary<log2>
+{
+    auto apply() const
+    {
+        const double ln2 = std::log(2.0);
+
+        return [ln2](auto x) { return std::log(x)/ln2; };
```
Comment (pfultz2) [inline src/include/migraphx/op/log2.hpp:41]: Use `std::log2` instead.

File: src/include/migraphx/op/log2.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct log2 : unary<log2>
+{
+    auto apply() const
+    {
+
+        return [](auto x) { return std::log2(x); };
```
Comment (umangyadav) [inline src/include/migraphx/op/log2.hpp:40]: add verify tests

**Review by umangyadav — APPROVED** [review]

File: src/include/migraphx/op/log2.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (shivadbhavsar) [inline src/include/migraphx/op/log2.hpp:4]:
```suggestion
 * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
```
CI probably failing due to this

**Review by shivadbhavsar — APPROVED** [review]

**Comment by shivadbhavsar** [comment]
I think you also have to add this op in `src/CMakeLists.txt` in `register_migraphx_ops`

File: src/include/migraphx/op/log2.hpp (hunk 2 above)
Comment (pfultz2) [reply src/include/migraphx/op/log2.hpp:40]: We should also have ref unit test added for this as well.

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**